13_07_2017/3D_slab_modes.py

```diff
@@ -0,0 +1,779 @@
+
+#import pdb # pause code for debugging at pdb.set_trace()
+import numpy as np
+import toolbox as tool
+import slab_functions as sf
+from pysac.plot.mayavi_seed_streamlines import SeedStreamline
+import matplotlib.pyplot as plt
+from mayavi import mlab
+import gc
+#import move_seed_points as msp
+import mayavi_plotting_functions as mpf
+import dispersion_diagram
+import img2vid as i2v
+
+###############################################################################
+
+# What mode do you want? OPTIONS:
+mode_options = ['slow-kink-surf', 'slow-saus-surf', 'slow-saus-body-3',
+                'slow-kink-body-3', 'slow-saus-body-2', 'slow-kink-body-2', 
+                'slow-saus-body-1', 'slow-kink-body-1', 'fast-saus-body-1',
+                'fast-kink-body-1', 'fast-saus-body-2', 'fast-kink-body-2',
+                'fast-saus-body-3', 'fast-kink-body-3', 'fast-kink-surf',
+                'fast-saus-surf', 'shear-alfven', 'shear-alfven-broadband']
+
+# Create sublists of modes
+alfven_mode_options = []
+kink_mode_options = []
+saus_mode_options = []
+slow_surf_mode_options = []
+fast_surf_mode_options = []
+fast_kink_mode_options = []
+fast_saus_mode_options = []
+slow_body_1_mode_options = []
+slow_body_2_mode_options = []
+slow_body_3_mode_options = []
+slow_body_mode_options = []
+fast_body_1_mode_options = []
+fast_body_2_mode_options = []
+fast_body_3_mode_options = []
+
+for mode in mode_options:
+    if 'alfven' in mode:
+        alfven_mode_options.append(mode)
+    if 'kink' in mode:
+        kink_mode_options.append(mode)
+    if 'saus' in mode:
+        saus_mode_options.append(mode)
+    if 'slow' in mode and 'surf' in mode:
+        slow_surf_mode_options.append(mode)
+    if 'fast' in mode and 'surf' in mode:
+        fast_surf_mode_options.append(mode)
+    if 'fast' in mode and 'kink' in mode:
+        fast_kink_mode_options.append(mode)
+    if 'fast' in mode and 'saus' in mode:
+        fast_saus_mode_options.append(mode)
+    if 'fast' in mode and 'body-1' in mode:
+        fast_body_1_mode_options.append(mode)
+    if 'fast' in mode and 'body-2' in mode:
+        fast_body_2_mode_options.append(mode)
+    if 'fast' in mode and 'body-3' in mode:
+        fast_body_3_mode_options.append(mode)
+    if 'slow' in mode and 'body-1' in mode:
+        slow_body_1_mode_options.append(mode)
+    if 'slow' in mode and 'body-2' in mode:
+        slow_body_2_mode_options.append(mode)
+    if 'slow' in mode and 'body-3' in mode:
+        slow_body_3_mode_options.append(mode)
+    if 'slow' in mode and 'body' in mode:
+        slow_body_mode_options.append(mode)
```

Comment:
It may or may not be easier/nicer to do this block with nested `if`s rather than a big long string of them (assuming that you still need to do this if you change it to use classes).